pilot/pkg/model/push_context.go

```diff
@@ -985,6 +1004,33 @@ func (ps *PushContext) initDestinationRules(env *Environment) error {
 	return nil
 }
 
+// AuthenticationPolicyForWorkload returns the matching auth policy for a given service
+// This replaces store.AuthenticationPolicyForWorkload
+func (ps *PushContext) AuthenticationPolicyForWorkload(service *Service, l labels.Instance, port *Port) *authn.Policy {
```

Comment:
how are you intending to call this? the plugin code needs to be updated. pass the proxy as is and use the proxies' labels inside the function. Try to maintain same signature style as other functions in this push context.. 
How do you even know the service for a workload? from service instances?

Reply:
why not simply return array of authN policies indexed by hostname and port?
AuthenticationPolicyForWorkload(proxy)

Reply:
@rshriram you can see how it is being called in my original PR here: https://github.com/istio/istio/pull/16296 
specifically https://github.com/istio/istio/blob/2da487fb952595210169e836d5990fe388b13416/pilot/pkg/networking/core/v1alpha3/cluster.go#L158
regarding the function signature for `AuthenticationPolicyForWorkload`-  I think I misinterpreted one of your earlier comments about the name of this function.  I thought you meant keep it in sync with the signature in `store.AuthenticationPolicyForWorkload`, rather than just keeping the names the same

Reply:
As described in the other comment, probably leave the label out for now.